DxDispatch/src/dxdispatch/OnnxDispatchable.cpp

```diff
@@ -135,68 +156,104 @@ void OnnxDispatchable::Bind(const Bindings& bindings)
+                            auto resource = m_device->CreateDefaultBuffer(DMLCalcBufferTensorSize(
+                                OnnxParsers::ConvertOnnxTensorDataType(tensorDataType),
+                                sizes.size(),
+                                sizes.data(),
+                                nullptr
+                                ));
+
+                            // Wrap the explicitly allocated DX resource.
+                            Microsoft::WRL::ComPtr<IUnknown> resourceWrapper;
+                            m_tensors.emplace_back(CreateTensorFromResource(
+                                m_ortDmlApi,
+                                dmlMemoryInformation,
+                                resource.Get(),
+                                tensorShape,
+                                tensorDataType,
+                                &resourceWrapper));
+
+                            m_tensorWrappers.push_back(std::move(resourceWrapper));
+                        }
+                        else
+                        {
+                            auto allocator = static_cast<OrtAllocator*>(Ort::AllocatorWithDefaultOptions());
+                            m_tensors.emplace_back(Ort::Value::CreateTensor(
+                                allocator, 
+                                tensorShape.data(),
+                                tensorShape.size(), 
+                                tensorDataType));
+                        }
+                    }
                 }
                 else
                 {
-                    m_ioBindings->BindOutput(tensorName.c_str(), m_tensors.back());
+                    // Wrap the pre-allocated DX resource.
+                    Microsoft::WRL::ComPtr<IUnknown> resourceWrapper;
+                    m_tensors.emplace_back(CreateTensorFromResource(
+                        m_ortDmlApi,
+                        dmlMemoryInformation,
+                        GetResourceFromModelBinding(tensorName, bindings),
+                        tensorShape,
+                        tensorDataType,
+                        &resourceWrapper));
+
+                    m_tensorWrappers.push_back(std::move(resourceWrapper));
+                }
+            }
+
+            if (isInputTensor)
+            {
+                if (typeInfo.GetONNXType() != ONNXType::ONNX_TYPE_TENSOR)
+                {
+                    // Don't bind non-tensor inputs.
+                    continue;
                 }
+
+                // Bind the input tensor.
+                m_ioBindings->BindInput(tensorName.c_str(), m_tensors.back());
             }
             else
             {
                 assert(!isInputTensor);
 
-                bool isDmlSupportedType = false;
-
-                if (typeInfo.GetONNXType() == ONNXType::ONNX_TYPE_TENSOR)
+                if (bindings.find(tensorName) == bindings.end())
```

Comment:
>bindings.find(tensorName) == bindings.end() [](http://example.com/codeflow?start=20&length=43)

DxDispatch uses C++ 20, right?

`if (!bindings.contains(tensorName))`

Reply:
C++17 (because of WSL/clang).